File: services/kg-jsonld-node/routes/jsonld.js
/*
 * Copyright 2020 EPFL/Human Brain Project PCO
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

var express = require('express');
const jsonld = require('jsonld');
var router = express.Router();
router.use(express.json());

router.post('/', function (req, res) {
  jsonld.expand(req.body, function (expandErr, expanded) {
    if (expandErr == null) {
      jsonld.compact(expanded, {}, function (compactErr, compacted) {
        if (compactErr == null) {
          res.send(compacted);
        } else {
          res.status(400);
          res.send("Was not able to compact JSON-LD", compactErr);
        }
      });
    } else {
      res.status(400);
      res.send({"error": "Was not able to expand JSON-LD", "detail": expandErr});
    }
  });
});

router.get('/health', function (req, res) {
  var payload = {
    "status": "UP"
  };
  res.status(200);
  res.send(payload);
});


module.exports = router;

File: services/kg-core-api/src/main/java/eu/ebrains/kg/core/api/JsonLD.java
/*
 * Copyright 2018 - 2021 Swiss Federal Institute of Technology Lausanne (EPFL)
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0.
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 *
 * This open source software code was developed in part or in whole in the
 * Human Brain Project, funded from the European Union’s Horizon 2020
 * Framework Programme for Research and Innovation under
 * Specific Grant Agreements No. 720270, No. 785907, and No. 945539
 * (Human Brain Project SGA1, SGA2 and SGA3).
 */

package eu.ebrains.kg.core.api;

import eu.ebrains.kg.commons.Version;
import eu.ebrains.kg.commons.api.JsonLd;
import eu.ebrains.kg.commons.config.openApiGroups.Simple;
import eu.ebrains.kg.commons.jsonld.JsonLdDoc;
import eu.ebrains.kg.commons.jsonld.NormalizedJsonLd;
import eu.ebrains.kg.commons.markers.ExposesInputWithoutEnrichedSensitiveData;
import io.swagger.v3.oas.annotations.Operation;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

/**
 * The spaces API provides information about existing KG spaces
 */
@RestController
@RequestMapping(Version.API + "/jsonld")
public class JsonLD {
    private final JsonLd.Client jsonLd;

    public JsonLD(JsonLd.Client jsonLd) {
        this.jsonLd = jsonLd;
    }

    @Operation(summary = "Normalizes the passed payload according to the EBRAINS KG conventions")
    @PostMapping("/normalizedPayload")
    @ExposesInputWithoutEnrichedSensitiveData
    @Simple
    public NormalizedJsonLd normalizePayload(@RequestBody JsonLdDoc payload) {
        return jsonLd.normalize(payload, true);
    }

}
